### PR TITLE
[Kotlin Migration] MaxExecFunction 

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] MaxExecFunction

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt [hunk 1]
```diff
@@ -14,54 +14,39 @@
...
     /**
      * Execute the faction, if it didn't get called for the same reference before and the number execution hasn't been reached
      *
      * @param ref the reference
      */
-    public void execOnceForReference(Object ref) {
-        if (mLastExecutionReference != null && mLastExecutionReference.get() == ref) {
-            return;
+    fun execOnceForReference(ref: Any) {
+        if (mLastExecutionReference != null && mLastExecutionReference!!.get() === ref) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt:46]: Needs a KotlinCleanup: thread safety on this class needs investigation

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
nit, conversion was good, but code was questionable

File: AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt:46]: @codingtosh annotate this then good to go!

**Review by mikehardy — CHANGES_REQUESTED** [review]
one annotation needed then merge

File: AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt (hunk 1 above)
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/utils/MaxExecFunction.kt:47]: @david-allison-1 @mikehardy  Done, PTAL.

**Review by mikehardy — APPROVED** [review]

**Comment by mikehardy** [comment]
David's only comment was the annotation, I think we're good

**Pull request merged**